Nytt panel for bekreftselse av vilkår

chfinst opened the pull request:
Ligner veldig på CheckboksPanel bare med orange/grønn bakgrunn avhengig av om den er checked eller ikke.

![checked](https://user-images.githubusercontent.com/14904827/37832040-bda58b4c-2ea7-11e8-8419-c67d99227673.PNG)
![not-checked](https://user-images.githubusercontent.com/14904827/37832041-bdcd096a-2ea7-11e8-8294-7c19807e06fd.PNG)

chfinst committed:
Nytt panel for bekreftselse av vilkår

Ligner veldig på CheckboksPanel bare med orange/grønn bakgrunn avhengig av om den er checked eller ikke.

chfinst committed:
Nytt panel for bekreftselse av vilkår

Ligner veldig på CheckboksPanel bare med orange/grønn bakgrunn avhengig av om den er checked eller ikke.

erlendev commented on packages/node_modules/nav-frontend-skjema/sample/_bekreftcheckbokspanel.sample.js:
```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line
```
Fint eksempel, men tenker kanskje vi kanskje får enda mer utviklerdok utifra å ikke bruke `no-transpilation.js`-delen av APIet her?

Forslag:
```
export default generateSample({
    baseType: BekreftCheckboksPanelExample,
    children: (<span>Her kan du sende inn html. For eksempel en <a href="/lenke">lenke</a></span>),
    attrs: {
        label: 'Jeg godtar',
        onChange: () => {}
    }
})
```

Foreslår også å droppe å bruke `<p>` og heller bare ta en `<span>` for å slippe den litt uheldige margin-en over/under.

nutgaard replied:
`BekreftCheckboksPanelExample` er en dum-komponent, så om vi skal vise frem funksjonaliteten så må vi nok bruke `no-transpilation`-delen. Evt. må vi eksponerer `checked` som en prop til appen, slik at man kan huke av en checkbox for å toggle visningen.

chfinst replied:
Har oppdatert koden med en span i eksempelet. La også til en liten margin under innholdet.

erlendev commented on packages/node_modules/nav-frontend-skjema/src/bekreft-checkboks-panel.tsx:
```diff
@@ -0,0 +1,100 @@
+        });
+    }
+
+    toggleHover(isHover: boolean) {
```
Trenger ikke et parameter for å styre denne tenker jeg, sett heller bare til `!this.state.isHover`. :)

nutgaard replied:
Basert på bruken bør den nok heller renames imho, e.g 
```
setHover(isHover: boolean) {
  return () => this.setState({ isHover });
}
```

Da kan den også brukes sånn `onMouseEnter={this.setHover(true)}`

chfinst replied:
Har oppdatert koden slik som Nicklas foreslår. Grunnen til at jeg ikke benyttet !this.state.isHover var at jeg ikke var helt sikker på hva dom skjer dersom musen ligger over komponenten når siden lastes. Er ikke sikkert det er noe problem.

Lillebo replied:
Er litt i tvil om hele panelet bør få en hover-stil på den måten når det bare er checkboksen og labelen som er klikkbar. Antar du har løst det slik pga. [designet](https://zpl.io/aXGgR0l) som ligger i Zeplin, men lurer litt på om det designet ble laget med en antagelse om at hele panelet skulle være klikkbar. Er det slik dere har løst det også @erlendev?

Skal høre med Sergio hva han mener om dette. Ellers knallbra PR :)

chfinst replied:
Ble løst slik med tanke på designet slik det er i Zeplin  ja @Lillebo. Om det bare er checkboks og label som skal ha hover så slipper man også å bruke javascript til å sette hover. Si fra når du har snakket med Sergio, så endrer vi på det. Kan også tenkes at han mener at innholdet i øvre del skal være alignet med starten på checkboksen :)

Lillebo replied:
Snakket med Sergio og vi ble enige om å begrense hover-stilen til bare checkboksen og label. La samtidig til en liten transition på panelet for å fade fra orange til grønn. Måtte gjøre endringene for å vise ham hvordan det ville bli, så committer like greit disse. Fint om du tar en titt @chfinst :)

Han vil også gjerne at innholdet i øvre del alignes med starten på checkboksen ja. Det har jeg ikke fikset.

chfinst replied:
Flott! Har aligna innholdet i øvre del med starten på checkboksen nå. Løste det ved å legge på -2rem i margin, hva tror du om det @Lillebo ?

La også til litt padding rundt label-teksten for å få litt større klikkflate.

Lillebo replied:
Råbra 👍

erlendev commented on packages/node_modules/nav-frontend-skjema/sample/_bekreftcheckbokspanel.sample.no-transpilation.js:
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
Denne fila kan trolig fjernes gitt at man går over til vanlig kodeeksempel som foreslått i kommentar over.

erlendev reviewed (changes requested):
Takk for fin PR! 👍 La inn noen kommentarer i koden. Det ser ut til at bygget her feiler pga. noe linting (se `npm run lint`), det må også fikses før vi kan merge.

Kobler på @sergiohaisch her, ettersom jeg er litt usikker på om contentet som legges inn som `children` bør paddes på lik linje med contentet ellers. Personlig tenker jeg at teksten øverst bør være på linje med starten av checkboxen. 

Kan du gi litt input her @sergiohaisch?

Lillebo committed:
Removed hover style on panel

Lillebo committed:
Merge branch 'master' of https://github.com/navikt/nav-frontend-moduler into bekreft-checkboks-panel

chfinst committed:
Merge remote-tracking branch 'origin/bekreft-checkboks-panel' into bekreft-checkboks-panel

# Conflicts:
#	packages/node_modules/nav-frontend-skjema-style/src/bekreft-checkboks-panel.less
#	packages/node_modules/nav-frontend-skjema/sample/_bekreftcheckbokspanel.sample.no-transpilation.js
#	packages/node_modules/nav-frontend-skjema/src/bekreft-checkboks-panel.tsx

chfinst committed:
Aligner innholdet i BekreftCheckboksPanel med checkboksen.

Legger også til litt padding rundt label slik at det blir litt større trykkflate.

Lillebo committed:
Fixed duplicate label class, label hitbox

Lillebo committed:
Lint fix
